src/main/java/org/wildfly/security/auth/client/SetCredentialStoreAuthenticationConfiguration.java

```diff
@@ -0,0 +1,99 @@
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.auth.client;
+
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+
+import org.wildfly.common.Assert;
+import org.wildfly.security.auth.callback.CredentialStoreCallback;
+import org.wildfly.security.credential.store.CredentialStore;
+import org.wildfly.security.credential.store.CredentialStoreException;
+import org.wildfly.security.credential.store.impl.KeystorePasswordStore;
+
+/**
+ * Subset of {@link AuthenticationConfiguration} handling {@link CredentialStore} configuration.
+ *
+ * @author <a href="mailto:pskopek@redhat.com">Peter Skopek</a>
+ */
+class SetCredentialStoreAuthenticationConfiguration extends AuthenticationConfiguration {
+
+    private final String name;
+    private final String type;
+    private final Map<String, String> attributes;
+
+    SetCredentialStoreAuthenticationConfiguration(AuthenticationConfiguration parent, String name, String type, Map<String, String> attributes) {
+        super(parent);
+        Assert.checkNotNullParam("name", name);
+        this.name = name;
+        this.type = type;
+        this.attributes = attributes;
+    }
+
+    @Override
+    AuthenticationConfiguration reparent(AuthenticationConfiguration newParent) {
+        return new SetCredentialStoreAuthenticationConfiguration(newParent, name, type, attributes);
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public Map<String, String> getAttributes() {
+        return attributes;
+    }
+
+    @Override
+    void handleCallback(final Callback[] callbacks, final int index) throws IOException, UnsupportedCallbackException {
+        final Callback callback = callbacks[index];
+        if (callback instanceof CredentialStoreCallback) {
+            return;
+        }
+        super.handleCallback(callbacks, index);
+    }
+
+    @Override
+    void handleCallbacks(final AuthenticationConfiguration config, final Callback[] callbacks) throws IOException, UnsupportedCallbackException {
+        CredentialStore credentialStore = null;
+        for (Callback callback : callbacks) {
+            if (callback instanceof CredentialStoreCallback) {
```

Comment:
We should not have a special callback for this; it should just use CredentialCallback and provide the credential as configured directly.